src/VirtualClient/VirtualClient.Actions/HadoopTerasort/HadoopMetricsParser.cs

```diff
@@ -0,0 +1,163 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace VirtualClient.Actions
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Data;
+    using System.Linq;
+    using System.Text.RegularExpressions;
+    using global::VirtualClient;
+    using global::VirtualClient.Contracts;
+    using DataTableExtensions = global::VirtualClient.Contracts.DataTableExtensions;
+
+    /// <summary>
+    /// Parser for Hadoop output document
+    /// </summary>
+    public class HadoopMetricsParser : MetricsParser
+    {
+        /// <summary>
+        /// Sectionize the text by one or more empty lines.
+        /// </summary>
+        private static readonly Regex HadoopSectionDelimiter = new Regex(@"(\n)(\s)*(\n)", RegexOptions.ExplicitCapture);
+
+        /// <summary>
+        /// Separate the column values using the equals to delimeter.
+        /// </summary>
+        private static readonly Regex HadoopDataTableDelimiter = new Regex(@"=", RegexOptions.ExplicitCapture);
+
+        /// <summary>
+        /// To separate value/unit like '1.86 GB/sec'. This regex looks forward for digit and backward for word.
+        /// </summary>
+        // private static readonly Regex ValueUnitSplitRegex = new Regex(@"^(.*?)(?:\(([^)]*)\))?$", RegexOptions.None);
+        private static readonly Regex ValueUnitSplitRegex = new Regex(@"^(.*?)(?:\s*\(([^)]*)\)\s*(.*))?$", RegexOptions.None);
```

Comment:
Just FYI we have a Regex for double etc in the TextParsing extensions if you need to use them